clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/CTInAppNativeInterstitialFragment.java

```diff
@@ -296,144 +280,116 @@ void cleanup() {
+        fullScreenIcon.setImageDrawable(ContextCompat.getDrawable(this.context, R.drawable.ct_ic_fullscreen_expand));
     }
 
-    private void disableFullScreenButton() {
-        fullScreenIcon.setVisibility(View.GONE);
-    }
+    private void openFullscreenDialog() {
+        View playerView = handle.videoSurface();
 
-    private void initFullScreenDialog() {
-        fullScreenDialog = new Dialog(this.context, android.R.style.Theme_Black_NoTitleBar_Fullscreen) {
-            public void onBackPressed() {
-                if (exoPlayerFullscreen) {
-                    closeFullscreenDialog();
+        imageViewLayoutParams = fullScreenIcon.getLayoutParams();
+        handle.switchToFullScreen(true);
+
+        // clear views from container
+        videoFrameLayout.removeView(playerView);
+        videoFrameLayout.removeView(fullScreenIcon);
+
+        if (fullScreenDialog == null) {
+            // create full screen dialog and show
+            fullScreenDialog = new Dialog(this.context, android.R.style.Theme_Black_NoTitleBar_Fullscreen) {
+                public void onBackPressed() {
+                    if (exoPlayerFullscreen) {
+                        closeFullscreenDialog();
+                    }
+                    super.onBackPressed();
                 }
-                super.onBackPressed();
-            }
-        };
-    }
+            };
+        }
 
-    private void openFullscreenDialog() {
-        imageViewLayoutParams = fullScreenIcon.getLayoutParams();
-        playerViewLayoutParams = playerView.getLayoutParams();
-        videoFramelayoutParams = videoFrameLayout.getLayoutParams();
-        ((ViewGroup) playerView.getParent()).removeView(playerView);
-        ((ViewGroup) fullScreenIcon.getParent()).removeView(fullScreenIcon);
-        ((ViewGroup) videoFrameLayout.getParent()).removeView(videoFrameLayout);
-        fullScreenDialog.addContentView(playerView, new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
+        ViewGroup.LayoutParams fullScreenParams = new ViewGroup.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT
+        );
+        fullScreenDialog.addContentView(playerView, fullScreenParams);
         exoPlayerFullscreen = true;
         fullScreenDialog.show();
     }
 
     private void playMedia() {
-        playerView.requestFocus();
-        playerView.setVisibility(View.VISIBLE);
-        playerView.setPlayer(player);
-        player.setPlayWhenReady(true);
+        handle.play();
     }
 
     private void prepareMedia() {
+        handle.initPlayerView(
+                context,
+                inAppNotification.isTablet() && isTablet()
+        );
+        streamContainer();
+
+        handle.initExoplayer(
+                context,
+                inAppNotification.getMediaList().get(0).getMediaUrl()
+        );
+    }
+
+    private void streamContainer() {
```

Comment:
Can be renamed to showStreamContainer()?

Reply:
done